wasm_utils/src/api_serialization/get_entry.rs

```diff
@@ -72,3 +87,82 @@ impl EntryHistory {
         }
     }
 }
+
+#[derive(Deserialize, Debug, Serialize, DefaultJson)]
+pub struct GetEntryResult {
+    pub found: bool, // indicates search was successful
+    // pub header: Option<ChainHeader>,   // header if requested in options
+    // pub sources: Option<Vec<Address>>, // sources if requested in options
+    pub history: Option<EntryHistory>,
+}
+impl GetEntryResult {
+    pub fn new() -> Self {
+        GetEntryResult {
+            found: false,
+            //            sources: None,
+            history: None,
+        }
+    }
+    pub fn found(&self) -> bool {
+        self.found
+    }
+
+    fn add_history(&mut self) {
+        self.history = Some(EntryHistory::new())
+    }
+
+    pub fn push(&mut self, entry_with_meta: &EntryWithMeta) {
+        if !self.found {
+            self.found = true;
+            self.add_history();
+        }
+        match self.history {
+            None => unreachable!(),
+            Some(ref mut history) => history.push(entry_with_meta),
+        };
```

Comment:
Is the `None` state really unreachable? Then I would advise to use `.expect("something went wrong")` and `.map` on the `self.history`.

If this shouldn't crash, I'll use an `if let Some(foo) = ...`